File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/store/MenuCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.store;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.Builder;

@TestFixture
public class MenuCreator {

    @Builder
    public static Menu create(
        @NotNull Store store,
        @NotNull String name,
        @NotNull String price,
        @NotNull MenuCategoryType category
    ) {
        return Menu.builder()
            .store(store)
            .name(name)
            .price(price)
            .category(category)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/store/MenuCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.store

import com.depromeet.threedollar.common.type.UserMenuCategoryType
import com.depromeet.threedollar.domain.rds.domain.userservice.store.Menu
import com.depromeet.threedollar.domain.rds.domain.userservice.store.Store
import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object MenuCreator {

    @JvmStatic
    fun create(
        store: Store,
        name: String,
        price: String,
        category: UserMenuCategoryType,
    ): Menu {
        return Menu.builder()
            .store(store)
            .name(name)
            .price(price)
            .category(category)
            .build()
    }

}
